### ISSUE TITLE
How to speed up `val` process ?

### ISSUE THREAD

**Issue by PapaMadeleine2022** [OML-Team/open-metric-learning#375]
Hello! After I trained my model, I would use this model to extract features by `[validate_cub.py]` like the method that I used [here](https://github.com/OML-Team/open-metric-learning/issues/374) for a hug amount of images. But there are limited GPUs. So how to speed up this `val` process?

**Comment by AlekseySh** [comment]
I did not get what you want.

If you want to speed up the validation process by using several GPUs -- the functionality is already here, nothing to speed up.
If you want to extract features from the images (just inference, no validation), there is no special script for this yet. But you can work on it by yourself, if you need an easy multi-GPU setup, just check `Trainer().predict()` from Lightning

**Comment by PapaMadeleine2022** [comment]
> I did not get what you want.
> 
> If you want to speed up the validation process by using several GPUs -- the functionality is already here, nothing to speed up. If you want to extract features from the images (just inference, no validation), there is no special script for this yet. But you can work on it by yourself, if you need an easy multi-GPU setup, just check `Trainer().predict()` from Lightning


I mean how to do validation on GPUs more faster ?

**Comment by AlekseySh** [comment]
Please, describe your case in numbers (how many images, what is the size of the query and gallery).
And did you do profiling to understand what part of the calculations is the heaviest one?

**Comment by AlekseySh** [comment]
Hey, @PapaMadeleine2022 !

I've made a Prediction pipeline, so you can perform inference on several GPUs by using a simple config. 
Just open the [features_extraction](https://github.com/OML-Team/open-metric-learning/tree/main/pipelines/features_extraction) section and you will find it in docs & example folders (see `predict.py`).

**Comment by AlekseySh** [comment]
If you find the output format not convenient, let's discuss other options.
